### PR TITLE
feat(pallets/subpsace): convert stake data type into dmaps

### PR DESCRIPTION
**Pull request by aripiprazole**
Converts StakeFrom and StakeTo data into a new double storage map

### CODE REVIEW

File: pallets/subspace/src/staking.rs [hunk 1]
```diff
@@ -318,26 +319,38 @@ impl<T: Config> Pallet<T> {
...
-        StakeTo::<T>::get(key)
+        StakeTo::<T>::iter()
+            .filter_map(|(to, from, value)| {
+                if &from == key {
+                    Some((to, value))
+                } else {
+                    None
+                }
+            })
+            .collect()
```
Comment (saiintbrisson) [inline pallets/subspace/src/staking.rs:330]:
```suggestion
        StakeTo::<T>::iter_prefix(key).collect()
```

File: pallets/subspace/src/staking.rs [hunk 2]
```diff
@@ -318,26 +319,38 @@ impl<T: Config> Pallet<T> {
...
     }
 
     pub fn set_stake_to_vector(key: &T::AccountId, stake_to_vector: BTreeMap<T::AccountId, u64>) {
         if stake_to_vector.is_empty() {
-            StakeTo::<T>::remove(key);
+            StakeTo::<T>::remove_prefix(key, None);
         } else {
-            StakeTo::<T>::insert(key, stake_to_vector);
+            for (to, v) in stake_to_vector.into_iter() {
+                StakeTo::<T>::insert(to, key, v);
```
Comment (saiintbrisson) [inline pallets/subspace/src/staking.rs:338]:
```suggestion
            for (to, v) in stake_to_vector.into_iter() {
                StakeTo::<T>::insert(key, to, v);
```

File: pallets/subspace/src/staking.rs [hunk 3]
```diff
@@ -318,26 +319,38 @@ impl<T: Config> Pallet<T> {
...
     ) {
-        StakeFrom::<T>::insert(module_key, stake_from_vector);
+        for (from, value) in stake_from_vector.iter() {
+            StakeTo::<T>::insert(module_key, from, *value);
+        }
     }
 
     pub fn get_stake_from_vector(module_key: &T::AccountId) -> BTreeMap<T::AccountId, u64> {
-        StakeFrom::<T>::get(module_key).into_iter().collect::<BTreeMap<_, _>>()
+        StakeTo::<T>::iter_prefix(module_key).collect::<BTreeMap<_, _>>()
```
Comment (saiintbrisson) [inline pallets/subspace/src/staking.rs:353]:
```suggestion
        StakeFrom::<T>::iter_prefix(module_key).collect()
```

**Review by saiintbrisson — CHANGES_REQUESTED** [review]

**Comment by Supremesource** [comment]
Solved on global-stake-v2

**Pull request closed without merging**